### DISCUSSION TITLE
Can we add an option to disable "New Device Logged In" messages?

### DISCUSSION THREAD

**Discussion by gamedevsam** [dani-garcia/vaultwarden#5103] [Ideas]
I'm trying to stay on the free plan of SendGrid, and these emails in particular are really eating into my quota.

**Comment by Gerardv514** [comment]
Uhh have you poked around in the admin panel? There’s an option for that already under advanced. 


![IMG_7759](https://github.com/user-attachments/assets/23b1c0db-faa5-45f2-845c-84676bc50180)

**Reply by BlackDex** [reply]
That is not totally true. 
That option is to fall a login if the email was unable to be sent. 
There is no option to disable this, and i also see no point in doing so as that will only lower the security.

**Reply by Gerardv514** [reply]
Seems like it is a two part thing that happens when enabled. Reading what we see at face value, I interpret as: When enabled an email is required to be send, and if that fails then login attempt will fail; this only happens when it is a new device.

So I would expect, based on reading the below, when set to false when a user logs in, using a new device, an email will not be required to be sent, and therefore since no email is sent it doesn't matter if the email fails because its not sent anyway.

So does the wording of what this does need to change then?

![image](https://github.com/user-attachments/assets/46123c43-cfc8-40a0-8ce7-4c15836c20cb)

**Reply by BlackDex** [reply]
It still does send an email. Only the login will not fail if for some reason the mail server was not reachable.
This is useful if you want to prevent some hacker to somehow being able to disrupt the mail server Vaultwarden is using and then login using the credentials of the user and that way prevent the user from receiving an email that someone has logged-in via an unknown device.

So, it will always send a mail when a new device is logged-in, but it will only fail the login if that setting is set to `true` and sending that new device login mail didn't worked.

**Reply by Gerardv514** [reply]
> So, it will always send a mail when a new device is logged-in, but it will only fail the login if that setting is set to `true` and sending that new device login mail didn't worked.

So this question should be changed to “Fail logins when new device login email fails” = true login will fail = false login will not fail/will proceed 

I had no clue this is what this controls, and I have it set to true because I want an email for new devices. I’m pretty tech savvy and if o didn’t catch this the average joe won’t either; it should be clarified a bit better I think.

**Reply by BlackDex** [reply]
I think it is correct as it is. It says **Require new device emails**, which indicates it **Requires** a mail. Also, the long description is correct also. I do not see a need to change both of them, though maybe the long description might not be fully correct regarding grammar, but I'm not a native English speaking person.

**Reply by stefan0xC** [reply]
> I had no clue this is what this controls, and I have it set to true because I want an email for new devices. I’m pretty tech savvy and if o didn’t catch this the average joe won’t either; it should be clarified a bit better I think.

The description of that option is pretty clear about what it does https://github.com/dani-garcia/vaultwarden/blob/13f4b66e6224775f194d4fc4274e7e948becb3a0/src/config.rs#L571-L573 and has nothing to do with the question being asked here (disabling those emails from being sent in the first place).

**Reply by Gerardv514** [reply]
> > I had no clue this is what this controls, and I have it set to true because I want an email for new devices. I’m pretty tech savvy and if o didn’t catch this the average joe won’t either; it should be clarified a bit better I think.
> 
> The description of that option is pretty clear about what it does
> 
> https://github.com/dani-garcia/vaultwarden/blob/13f4b66e6224775f194d4fc4274e7e948becb3a0/src/config.rs#L571-L573
> 
> and has nothing to do with the question being asked here (disabling those emails from being sent in the first place).

Sure it’s clear in the code , if you read code. End users don’t read code and will read the ui, which does not say what is stated in the code

**Reply by Gerardv514** [reply]
And this does have to do with question here because I thought from ui perspective that this setting with either send a new device email when check and wont send a new device email when not checked.

**Reply by Gerardv514** [reply]
> I think it is correct as it is. It says **Require new device emails**, which indicates it **Requires** a mail. Also, the long description is correct also. I do not see a need to change both of them, though maybe the long description might not be fully correct regarding grammar, but I'm not a native English speaking person.

If it says require new device emails, and I uncheck it I would expect not to have email required (aka sent out) which does not include fail. From the perspective does an email go out yes or not. You’re seeing from perspective does it fail yes or no.

**Reply by Gerardv514** [reply]
Clarifying im seeing this check as relating to the first question yes or no to whether an email is sent. 

When the option is set to true means yes I want an email and if the email fails then login will fail. 

When option is set to false I’m reading it as no I don’t want the emails and therefore failing doesn’t matter because I said false to wanting these emails.

**Reply by BlackDex** [reply]
No, you said false to require an email, not opt-out of receiving them at all

**Reply by stefan0xC** [reply]
> Sure it’s clear in the code , if you read code. End users don’t read code and will read the ui, which does not say what is stated in the code

You yourself posted a screenshot of the description being shown in the UI. https://github.com/dani-garcia/vaultwarden/discussions/5103#discussioncomment-10992081

**Reply by Gerardv514** [reply]
> > Sure it’s clear in the code , if you read code. End users don’t read code and will read the ui, which does not say what is stated in the code
> 
> You yourself posted a screenshot of the description being shown in the UI. [#5103 (reply in thread)](https://github.com/dani-garcia/vaultwarden/discussions/5103#discussioncomment-10992081)

Well that part didn’t say an email is always sent out regardless of what is checked off; therefore I’m sure you would have to go somewhere else in code to find that bit of information. 

Either way now its clarified of what it does after this discussion. Thanks

**Comment by gamedevsam** [comment]
I understand this is counter to the developer's wishes. Closing this, thanks for the discussion!
